pkg/cmd/kafka/acl/grant/grant.go

```diff
@@ -0,0 +1,228 @@
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasKafka() {
+				return opts.localizer.MustLocalizeError("kafka.acl.common.error.noKafkaSelected")
+			}
+
+			opts.kafkaID = cfg.Services.Kafka.ClusterID
+
+			if opts.user == "" && opts.svcAccount == "" {
+				return opts.localizer.MustLocalizeError("kafka.acl.common.error.noPricipalsSelected")
+			}
+
+			if opts.user != "" && opts.svcAccount != "" {
+				return opts.localizer.MustLocalizeError("kafka.acl.common.error.bothPricipalsSelected")
+			}
+
+			return runGrantPermissions(opts)
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.user, "user", "", opts.localizer.MustLocalize("kafka.acl.grantPermissions.flag.user.description"))
+	cmd.Flags().StringVar(&opts.svcAccount, "service-account", "", opts.localizer.MustLocalize("kafka.acl.grantPermissions.flag.serviceAccount.description"))
+	cmd.Flags().StringVar(&opts.topic, "topic", "", opts.localizer.MustLocalize("kafka.acl.grantPermissions.flag.topic.description"))
+	cmd.Flags().StringVar(&opts.group, "group", "", opts.localizer.MustLocalize("kafka.acl.grantPermissions.flag.group.description"))
+	cmd.Flags().BoolVar(&opts.consumer, "consumer", false, opts.localizer.MustLocalize("kafka.acl.grantPermissions.flag.consumer.description"))
+	cmd.Flags().BoolVar(&opts.producer, "producer", false, opts.localizer.MustLocalize("kafka.acl.grantPermissions.flag.producer.description"))
+	cmd.Flags().BoolVar(&opts.prefix, "prefix", false, opts.localizer.MustLocalize("kafka.acl.grantPermissions.flag.prefix.description"))
+	cmd.Flags().BoolVar(&opts.admin, "admin", false, opts.localizer.MustLocalize("kafka.acl.grantPermissions.flag.admin.description"))
+
+	return cmd
+}
+
+// nolint:funlen
+func runGrantPermissions(opts *options) (err error) {
+
+	conn, err := opts.Connection(connection.DefaultConfigRequireMasAuth)
+	if err != nil {
+		return err
+	}
+
+	api, kafkaInstance, err := conn.API().KafkaAdmin(opts.kafkaID)
+	if err != nil {
+		return err
+	}
+
+	kafkaName := kafkaInstance.GetName()
+
+	var topicNameArg string = acl.Wildcard
+	var groupIdArg string = acl.Wildcard
+	var patternArg kafkainstanceclient.AclPatternType = kafkainstanceclient.ACLPATTERNTYPE_LITERAL
+
+	var userArg string
+
+	if opts.topic != "" {
+		topicNameArg = opts.topic
+	}
+
+	if opts.prefix {
+		patternArg = kafkainstanceclient.ACLPATTERNTYPE_PREFIXED
+	}
+
+	if opts.user != "" {
+		userArg = buildPrincipal(opts.user)
+	}
+
+	if opts.svcAccount != "" {
+		userArg = buildPrincipal(opts.svcAccount)
+	}
+
+	req := api.AclsApi.CreateAcl(opts.Context)
+
+	if opts.producer || opts.consumer {
+		aclBindTopicDescribe := *kafkainstanceclient.NewAclBinding(kafkainstanceclient.ACLRESOURCETYPE_TOPIC, topicNameArg, patternArg, userArg, kafkainstanceclient.ACLOPERATION_DESCRIBE, kafkainstanceclient.ACLPERMISSIONTYPE_ALLOW)
+
+		req = req.AclBinding(aclBindTopicDescribe)
+
+		err = acl.ExecuteACLRuleCreate(req, opts.localizer, kafkaName)
+		if err != nil {
+			return err
```

Comment:
That needs mapping to something more user-friendly. Backend errors by default will only include status code (no actual info what went wrong)